runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2721,6 +2721,22 @@ done:;
 		return EXECUTE_BYTECODE;
 	}
 
+	/* java.lang.Class: public native Class<?> arrayType(); */
+	VMINLINE VM_BytecodeAction
+	inlClassArrayType(REGISTER_ARGS_LIST)
+	{
+		j9object_t arrayObject = NULL;
+		J9Class *componentClazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, *(j9object_t*)_sp);
+		J9Class *arrayClazz = internalCreateArrayClass(_currentThread, 
+				(J9ROMArrayClass *) J9ROMIMAGEHEADER_FIRSTCLASS(_currentThread->javaVM->arrayROMClasses), 
+				componentClazz);
+		if (NULL != arrayClazz) {
+			arrayObject = J9VM_J9CLASS_TO_HEAPCLASS(arrayClazz);
+		}
```

Comment:
In the NULL case, there should be an exception pending, which requires a different return code. I suggest initializing an rc to `GOTO_THROW_CURRENT_EXCEPTION` and setting the rc to `EXECUTE_BYTECODE` in the success case (if the arrayClazz is non-NULL). The `returnObjectFromINL` should also only be performed in the success case.